ccm_web/server/src/canvas/canvas.service.ts

```diff
@@ -67,8 +67,13 @@ export class CanvasService {
       const response = await this.httpService.post<TokenResponseBody>(
         `${this.url}/login/oauth2/token?${searchParams.toString()}`
       ).toPromise()
-      data = response.data
-      logger.debug(JSON.stringify(data, null, 2))
+      logger.debug(`Status code: ${response.status}`)
+      if (response.status === HttpStatus.OK) {
+        data = response.data
+      } else {
+        logger.error(`Received unusual status code ${response.status}`)
+        logger.error(`Response body: ${JSON.stringify(response.data, null, 2)}`)
+      }
```

Comment:
I realized that we needed at least some basic error handling here for irregular status codes.